apps/library/Makefile

```diff
@@ -29,59 +32,73 @@ else
     BASE_CFLAGS += -g -DDEBUG
 endif
 
-DEFINES :=
+BASE_DEFINES :=
 
 ifeq ($(RV32E), 1)
-    DEFINES += -DUSE_RV32E=1
+    LIB_DEFINES += -DUSE_RV32E=1
 endif
 
 override CFLAGS := $(BASE_CFLAGS) $(DEFINES) $(CFLAGS)
-override LDFLAGS := -Tbuild/gpc_link.common.lds -nostdlib -m elf32lriscv $(LDFLAGS)
+override LDFLAGS := -Tout/$(PREFIX)/lib/libgpc_link.lds -nostdlib -m elf32lriscv $(LDFLAGS)
```

Comment:
Currently, the Use of "nostdlib" prevents your code to use standard GCC libs such as DIV, MULSI etc. 
you should try using "nostartfiles" instead of the "nostdlib".